plugins/modules/mso_schema_site_contract_service_graph_listener.py

```diff
@@ -0,0 +1,744 @@
+                if service_node.get("serviceNodeRef") and service_node.get("serviceNodeRef").split("/")[-1] == "node{0}".format(service_node_index + 1):
+                    listeners = service_node.get("deviceConfiguration", {}).get("cloudLoadBalancer", {}).get("listeners", [])
+                    if listeners:
+                        parent_present = True
+                        for listener_data in listeners:
+                            if listener_data.get("name") == listener:
+                                listeners_data.append(listener_data)
+                                break
+                else:
+                    mso.fail_json(
+                        msg="The service_node_index: {0} is not matching with the service node reference: {1}.".format(
+                            service_node_index, service_node.get("serviceNodeRef")
+                        )
+                    )
+            else:
+                mso.fail_json(msg="The service_node_index: {0} is out of range.".format(service_node_index))
+
+        # Query all listeners under a contract does not require service_node_index, so the below condition is required
+        elif state == "query":
+            # Query all listeners under a contract
+            for service_node in service_nodes:
+                listeners = service_node.get("deviceConfiguration", {}).get("cloudLoadBalancer", {}).get("listeners", [])
+                if listener:
+                    for listener_data in listeners:
+                        # Query a listener under a contract service node
+                        if listener_data.get("name") == listener:
+                            listeners_data.append(listener_data)
+                            break
+                else:
+                    listeners_data.extend(listeners)
+        else:
+            mso.fail_json(msg="The service_node_index: {0} is not valid.".format(service_node_index))
+
+    if len(listeners_data) == 1:
+        # Query a specific listener
+        mso.existing = listeners_data[0]
+    elif len(listeners_data) >= 1:
+        # Query all listeners under a contract
+        mso.existing = listeners_data
+
+    if state == "query":
+        mso.exit_json()
+
+    ops = []
+    mso.previous = mso.existing
+
+    parent_object = {}
+    listener_object = {}
+
+    if state == "present":
+        # Parent object creation logic begins
+        if device is None and parent_present is False:
+            mso.fail_json(msg="The 'device' name is required to initialize the parent object.")
+
+        elif device is not None and parent_present is False:
+            query_device_data = mso.lookup_service_node_device(site_id, tenant, device_name=device)
+
+            if (query_device_data.get("deviceVendorType") == "NATIVELB" and query_device_data.get("devType") == "application") or (
+                query_device_data.get("deviceVendorType") == "NATIVELB" and query_device_data.get("devType") == "network"
+            ):
```

Comment:
Is this achieving same result?

```suggestion
            if query_device_data.get("deviceVendorType") == "NATIVELB" and (query_device_data.get("devType") == "application") or query_device_data.get("devType") == "network"):
```